### PR TITLE
Validations and model specs

### PR DESCRIPTION
**Pull request by danielochuba**
# Blog app - Validations and model specs
### In this pull request I did the following: 
- Created model test for
    - `User`
    - `Post`
    - `Comment`
    - `Like`
- Established validations for 
    - For the User model:
         - Name must not be blank.
         - PostsCounter must be an integer greater than or equal to zero.
     - For the Post model:
          - Title must not be blank.
          - Title must not exceed 250 characters.
          - CommentsCounter must be an integer greater than or equal to zero.
          - LikesCounter must be an integer greater than or equal to zero.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -3,6 +3,9 @@ class User < ApplicationRecord
   has_many :comments
   has_many :likes
 
+  validates :name, presence: true
+  validates :post_count, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+
   def recent_posts(limit = 3)
     authored_posts.order(created_at: :desc).limit(limit)
   end
```
Comment (Owusu-Desmond) [inline app/models/user.rb:11]: - [ ] I suggest that any custom methods in the models should have a corresponding unit test so please make sure that you add a test for this `recent_posts` method :pray:

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  subject { Post.new(title: 'My Post', text: 'This is my post content') }
+
```
Comment (Owusu-Desmond) [inline spec/models/post_spec.rb:5]: - [ ] I suggest that any custom methods in the models should have a corresponding unit test so please ensure that you add a test for the `update_user_posts_counter` and `recent_comments` in your post model :pray:

File: spec/models/like_spec.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+require 'rails_helper'
+
+RSpec.describe Like, type: :model do
```
Comment (Owusu-Desmond) [inline spec/models/like_spec.rb:3]: - [ ] I suggest that any custom methods in the models should have a corresponding unit test so please ensure that you add a test for the `increment_post_likes_counter` method in your like model :pray:

File: spec/models/comment_spec.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
```
Comment (Owusu-Desmond) [inline spec/models/comment_spec.rb:3]: - [ ] I suggest that any custom methods in the models should have a corresponding unit test so please ensure that you add a test for the `increment_post_comments_counter` method in your comment model :pray:

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @danielochuba , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sumairq — APPROVED** [review]
### Hi @danielochuba , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good job implementing changes
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**